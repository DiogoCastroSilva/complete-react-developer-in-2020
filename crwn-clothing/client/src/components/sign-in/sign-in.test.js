import React from 'react';
import { shallow } from 'enzyme';

import SignIn from './sign-in.component';


describe('testing SignIn component', () => {
    let wrapper;
    let onEmailSignInStart;
    let onGoogleSignInStart;

    beforeEach(() => {
        onEmailSignInStart = jest.fn();
        onGoogleSignInStart = jest.fn();

        const mockProps = {
            onEmailSignInStart: onEmailSignInStart,
            onGoogleSignInStart: onGoogleSignInStart
        };

        wrapper = shallow(<SignIn {...mockProps} />);
    });

    it('should render a MenuItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});