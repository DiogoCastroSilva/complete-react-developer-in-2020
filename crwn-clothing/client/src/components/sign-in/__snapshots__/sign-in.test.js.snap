// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SignIn component should render a MenuItem component 1`] = `
<styled.div>
  <styled.h2>
    I alredy have an account
  </styled.h2>
  <span>
    Sign in with your email and password
  </span>
  <form
    onSubmit={[Function]}
  >
    <FormInput
      handleChange={[Function]}
      label="email"
      name="email"
      required={true}
      type="email"
      value=""
    />
    <FormInput
      handleChange={[Function]}
      label="passsword"
      name="password"
      required={true}
      type="password"
      value=""
    />
    <styled.div>
      <CustomButton
        type="submit"
      >
        Sign in
      </CustomButton>
      <CustomButton
        color="primary"
        onClick={[MockFunction]}
        type="button"
      >
        Sign in with Google
      </CustomButton>
    </styled.div>
  </form>
</styled.div>
`;
